### PR TITLE
Add an optional kind node image parameter to dev-env

### PR DESCRIPTION
**Pull request by fedepaol**
Being able to specify the image to use for the node allows us to create
test cluster pinned to specific kind versions.

### CODE REVIEW

**Comment by fedepaol** [comment]
cc @markdgray

File: tasks.py [hunk 1]
```diff
@@ -186,13 +186,14 @@ def validate_kind_version():
     "protocol": "Pre-configure MetalLB with the specified protocol. "
                 "Unconfigured by default. Supported: 'bgp','layer2'",
```
Comment (markdgray) [inline tasks.py:187]: Can you add a description of this option? I know it is in the general section below but I think convention would also put it in the `@task(help={` section

Comment (fedepaol) [reply tasks.py:187]: I missed that. Will do!

File: tasks.py [hunk 2]
```diff
@@ -216,11 +217,14 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None, protocol=None):
...
             }
+        extra_options = ""
+        if node_img != None:
+            extra_options = "--image={}".format(node_img)
         config = yaml.dump(config).encode("utf-8")
         with tempfile.NamedTemporaryFile() as tmp:
             tmp.write(config)
             tmp.flush()
-            run("kind create cluster --name={} --config={}".format(name, tmp.name), pty=True, echo=True)
+            run("kind create cluster --name={} --config={} {}".format(name, tmp.name, extra_options), pty=True, echo=True)
```
Comment (markdgray) [inline tasks.py:227]: Why did you add "extra_options" instead of just "--image={}"?

Comment (fedepaol) [reply tasks.py:227]: because otherwise it wouldn't work if we don't specify the image.
I am open to suggestions, my python-fu is pretty bad :-)

File: tasks.py (hunk 1 above)
Comment (markdgray) [reply tasks.py:187]: Thanks, an example image name would be useful as I am still trying to find an example to test.

Comment (fedepaol) [reply tasks.py:187]: You can find them here https://github.com/kubernetes-sigs/kind/releases
I tried and it seemed to work :P

File: tasks.py (hunk 2 above)
Comment (markdgray) [reply tasks.py:227]: oh that makes sense. Im just as bad but what you have done makes sense to me

File: tasks.py (hunk 1 above)
Comment (markdgray) [reply tasks.py:187]: you could add something like `(e.g. kindest/node:v1.18.19)`

Comment (fedepaol) [reply tasks.py:187]: yes will do

Comment (markdgray) [reply tasks.py:187]: Thanks, just tested it there and seems to work! cool.

Comment (fedepaol) [reply tasks.py:187]: Done

**Comment by markdgray** [comment]
This looks good to me as long as it passes the tests.

**Comment by russellb** [comment]
lgtm.  It would be good to add an example to dev-env/README.md, but don't need to block on it
